Python/optimizer.c

```diff
@@ -161,11 +162,24 @@ _PyOptimizer_BackEdge(_PyInterpreterFrame *frame, _Py_CODEUNIT *src, _Py_CODEUNI
         if (err < 0) {
```

Comment:
Should we assert the executor is `NULL` here? And maybe initialize it to `NULL` above, if so?